src-opam/distro.ml

```diff
@@ -1056,19 +672,20 @@ let tag_of_distro (d : t) =
   | `OpenSUSE `V15_5 -> "opensuse-15.5"
   | `OpenSUSE `Tumbleweed -> "opensuse-tumbleweed"
   | `OpenSUSE `Latest -> "opensuse"
-  | `Cygwin v -> "cygwin-" ^ win10_release_to_string v
-  | `Windows (`Mingw, v) -> "windows-mingw-" ^ win10_release_to_string v
-  | `Windows (`Msvc, v) -> "windows-msvc-" ^ win10_release_to_string v
+  | `Cygwin `Ltsc2016 -> "cygwin-2016"
+  | `Cygwin `Ltsc2019 -> "cygwin-2019"
+  | `Cygwin `Ltsc2022 -> "cygwin-2022"
+  | `Cygwin `Latest -> "cygwin-2022"
+  | `Windows (`Mingw, `Ltsc2019) -> "windows-mingw-ltsc2019"
+  | `Windows (`Mingw, `Latest) -> "windows-mingw-ltsc2019"
+  | `Windows (`Msvc, `Ltsc2019) -> "windows-msvc-ltsc2019"
+  | `Windows (`Msvc, `Latest) -> "windows-msvc-ltsc2019"
+  | `WindowsServer (`Mingw, `Ltsc2022) -> "windows-server-mingw-ltsc2022"
+  | `WindowsServer (`Mingw, `Latest) -> "windows-server-mingw-ltsc2022"
+  | `WindowsServer (`Msvc, `Ltsc2022) -> "windows-server-msvc-ltsc2022"
+  | `WindowsServer (`Msvc, `Latest) -> "windows-server-msvc-ltsc2022"
```

Comment:
The duplication here (especially w.r.t `` `Latest ``) I think is unfortunate. How about keeping `win10_release_to_string` but calling it something more like `windows_edition_to_string`:
```ocaml
let windows_edition_to_string = function
| `Ltsc2016 -> "ltsc2016"
| `Ltsc2019 -> "ltsc2019"
| `Ltsc2022 -> "ltsc2022"

let rec windows_edition_latest = function
| `Cygwin -> windows_edition_latest `WindowsServer
| `Windows -> `Ltsc2019
| `WindowsServer -> `Ltsc2022
```

Reply:
I agree that the duplication is unfortunate.  If we use `windows_edition_to_string`, how would we resolve <code>`Latest</code>?

Having reviewed the other platforms again, I wonder if these lines should actually read as follows, which generates additional tags.  It also highlighted that `resolve_alias` and `distro_of_tag` were not equally opposite.

```ocaml
  | `Cygwin `Ltsc2016 -> "cygwin-2016"
  | `Cygwin `Ltsc2019 -> "cygwin-2019"
  | `Cygwin `Ltsc2022 -> "cygwin-2022"
  | `Cygwin `Latest -> "cygwin"
  | `Windows (`Mingw, `Ltsc2019) -> "windows-mingw-ltsc2019"
  | `Windows (`Mingw, `Latest) -> "windows-mingw"
  | `Windows (`Msvc, `Ltsc2019) -> "windows-msvc-ltsc2019"
  | `Windows (`Msvc, `Latest) -> "windows-msvc"
  | `WindowsServer (`Mingw, `Ltsc2022) -> "windows-server-mingw-ltsc2022"
  | `WindowsServer (`Mingw, `Latest) -> "windows-server-mingw"
  | `WindowsServer (`Msvc, `Ltsc2022) -> "windows-server-msvc-ltsc2022"
  | `WindowsServer (`Msvc, `Latest) -> "windows-server-msvc"
```